File: sem2/ADF-I/java-1/d06_abstract/src/data/Rectangle.java
/*
 * mo ta cau truc 1 doi tuong hinh chu nhat, la con cua lop Shape
 * co 2 thuoc tinh chieu dai va chieu rong
 */
package data;

import java.util.Scanner;


public class Rectangle extends Shape {
    public int width, length;
    
    
    @Override
    public double getArea() {
        return width * length;
    }

    @Override
    public double getPeripheral() {
        return (width+length)*2;
    }
    
    @Override
    public void input(){
        Scanner sc = new Scanner(System.in);
        System.out.print("Nhap chieu rong: ");
        width = Integer.parseInt( sc.nextLine().trim());
        System.out.print("Nhap chieu dai: ");
        length = Integer.parseInt( sc.nextLine().trim());
    }

    @Override
    public void output() {
        System.out.println("Thong tin hinh chu nhat");
        super.output();
    }
    
    
}

File: sem2/ap-assignments/AP/d03_abstract/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace d03_abstract
{
    public class Rectangle : Shape
    {

        public float h;
        public float w;
        public void Input()
        {
            
            Console.Write("Nhap chieu dai:");
            h = float.Parse(Console.ReadLine().Trim());
            Console.Write("Nhap chieu rong:");
            w = float.Parse(Console.ReadLine().Trim());
        }
        public override float Area()
        {
            return w * h; 
        }

        public override float Peripheral()
        {
            return (w + h) / 2;
        }
    }
}
